geocity/apps/api/serializers.py

```diff
@@ -756,3 +753,207 @@ class SubmissionFiltersSerializer(serializers.Serializer):
+                    ]["values"] = category_value_list
+
+                # Properties for detailed API
+                else:
+                    result["properties"][
+                        field["field_values__field__api_name"]
+                    ] = field["field_values__value__val"]
+
+                # Custom way to transform a string "True" and "False" to True and False
+                # FIXME: Not a good practice, need to develop a real boolean field instead of string to bool
+                result["properties"][field["form__amend_fields__api_name"]] = (
+                    True
+                    if field["form__amend_fields__amend_field_value__value"] == "True"
+                    else False
+                    if field["form__amend_fields__amend_field_value__value"] == "False"
+                    else field["form__amend_fields__amend_field_value__value"]
+                )
+            # Light API for agenda
+            else:
+                # Field visible on light API and it's not used as filter
+                if (
+                    field["field_values__field__api_light"]
+                    and not field["field_values__field__used_as_api_filter"]
+                ):
+                    # Properties for light API
+                    result["properties"][
+                        field["field_values__field__api_name"]
+                    ] = field["field_values__value__val"]
+
+                # Amend field for light API
+                if field["form__amend_fields__api_light"]:
+
+                    # Store amend properties for light API
+                    result["properties"][field["form__amend_fields__api_name"]] = field[
+                        "form__amend_fields__amend_field_value__value"
+                    ]
+
+    # Custom way to retrieve starts_at and ends_at for both light and detailed
+    geo_time_qs = value.geo_time.all()
+
+    aggregated_geotime_qs = geo_time_qs.values("submission_id").aggregate(
+        submission_geo_time_start_date=Min("starts_at"),
+        submission_geo_time_end_date=Max("ends_at"),
+    )
+
+    local_tz = timezone(timedelta(hours=2))
+    result["properties"]["starts_at"] = (
+        aggregated_geotime_qs["submission_geo_time_start_date"]
+        .replace(tzinfo=timezone.utc)
+        .astimezone(local_tz)
+        if aggregated_geotime_qs["submission_geo_time_start_date"]
+        else ""
+    )
+    result["properties"]["ends_at"] = (
+        aggregated_geotime_qs["submission_geo_time_end_date"]
+        .replace(tzinfo=timezone.utc)
+        .astimezone(local_tz)
+        if aggregated_geotime_qs["submission_geo_time_end_date"]
+        else ""
+    )
+
+    # Rewrite poster to match agenda-embed
+    if "poster" in result["properties"]:
+        result["properties"]["poster"] = {
+            "src": f'http://localhost:9095/agenda/image/display/{result["properties"]["poster"]}',
+            "width": "1365",
+            "height": "2048",
```

Comment:
Define how we can have width and height ? Is it the actual image width and height ?

Reply:
This is not how it's working. I've been doing some tests, and this value is required to show an image. We can still resize the image through `Thumblor`. Seems like to be the original file dimensions